src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -251,6 +252,31 @@ public static User getUserContext(Client client) {
         return User.parse(userStr);
     }
 
+    /**
+     * Checks whether resource authz framework should be used.
+     * @param resourceType the type to check for resource authz
+     * @return true if resource-authz should be used, false otherwise
+     */
+    public static boolean shouldUseResourceAuthz(String resourceType) {
+        var client = ResourceSharingClientAccessor.getInstance().getResourceSharingClient();
+        return client != null && client.isFeatureEnabledForType(resourceType);
+    }
+
+    /**
+     * Verifies whether the user has permission to access the resource.
+     * @param resourceType the type of resource to be authorized
+     * @param onSuccess consumer function to execute if resource sharing feature is enabled
+     * @param fallBackIfDisabled consumer function to execute if resource sharing feature is disabled.
+     */
+    public static void verifyResourceAccessAndProcessRequest(String resourceType, Runnable onSuccess, Runnable fallBackIfDisabled) {
+        // Resource access will be auto-evaluated
+        if (shouldUseResourceAuthz(resourceType)) {
```

Comment:
Should be under a try/catch block here

Reply:
we don't need any all exceptions must be thrown up to the caller.